### PR TITLE
Refactor workspace

### PR DESCRIPTION
**Pull request by barbararochazup**
## Issue Description

The user interface is not up to date with the rules of the backend. Causing undesirable side effects.

## Solution

group all workspace calls at a single point, removing unnecessary data from the global state and updating the way permissions between workspaces and the user are handled.

### CODE REVIEW

File: moove/application/src/main/kotlin/io/charlescd/moove/application/usergroup/response/UserGroupResponse.kt [hunk 1]
```diff
@@ -27,6 +28,7 @@ data class UserGroupResponse(
     val author: SimpleUserResponse,
     @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
     val createdAt: LocalDateTime,
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
     val users: List<SimpleUserResponse>
```
Comment (pedroborgeszup) [inline moove/application/src/main/kotlin/io/charlescd/moove/application/usergroup/response/UserGroupResponse.kt:32]: To maintain the response pattern, I think should return this value, even if empty

File: moove/application/src/main/kotlin/io/charlescd/moove/application/workspace/response/SimpleWorkspaceResponse.kt [hunk 2]
```diff
@@ -16,8 +16,27 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude
+import io.charlescd.moove.domain.SimpleWorkspace
+
 data class SimpleWorkspaceResponse(
     val id: String,
     val name: String,
-    val permissions: List<String>
-)
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val permissions: List<String>,
```
Comment (pedroborgeszup) [inline moove/application/src/main/kotlin/io/charlescd/moove/application/workspace/response/SimpleWorkspaceResponse.kt:26]: To maintain the response pattern, I think should return this value, even if empty

**Review by pedroborgeszup — CHANGES_REQUESTED** [review]

File: ui/src/modules/Users/__tests__/hooks.spec.ts [hunk 3]
```diff
@@ -146,36 +146,6 @@ test('should throw an error in useUser', async () => {
...
-  let response: Promise<User>;
-
-  await act(async () => {
-    response = await current.findWorkspacesByUser(user.id);
-  });
-
-  expect(response).toBeUndefined();
-});
-
 // TODO hooks.spec.tsx to .ts
```
Comment (luanecavalcantizup) [inline ui/src/modules/Users/__tests__/hooks.spec.ts:149]: remove comments

File: moove/application/src/main/kotlin/io/charlescd/moove/application/usergroup/response/UserGroupResponse.kt (hunk 1 above)
Comment (barbararochazup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/usergroup/response/UserGroupResponse.kt]: Done

File: moove/application/src/main/kotlin/io/charlescd/moove/application/workspace/response/SimpleWorkspaceResponse.kt (hunk 2 above)
Comment (barbararochazup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/workspace/response/SimpleWorkspaceResponse.kt:25]: Done

**Review by luanecavalcantizup — DISMISSED** [review]

**Review by pedroborgeszup — APPROVED** [review]

**Review by luanecavalcantizup — DISMISSED** [review]

**Review by pedroborgeszup — APPROVED** [review]

File: ui/src/core/utils/abilities.ts [hunk 4]
```diff
@@ -37,6 +37,8 @@ const setUserAbilities = () => {
   const permissions = getPermissions();
   const { can, rules } = new AbilityBuilder<AppAbility>();
 
+  console.log('setUserAbilities: isRoot()', isRoot());
```
Comment (luanecavalcantizup) [inline ui/src/core/utils/abilities.ts]: remove console.log

**Review by luanecavalcantizup — APPROVED** [review]

**Review by pedroborgeszup — DISMISSED** [review]

**Pull request closed without merging**